test/support/src/vfs_helpers.cc

```diff
@@ -451,12 +451,16 @@ VFSTestBase::VFSTestBase(
 }
 
 VFSTestBase::~VFSTestBase() {
-  if (vfs_.supports_uri_scheme(temp_dir_)) {
-    bool is_dir = false;
-    vfs_.is_dir(temp_dir_, &is_dir).ok();
-    if (is_dir) {
-      vfs_.remove_dir(temp_dir_).ok();
+  try {
+    if (vfs_.supports_uri_scheme(temp_dir_)) {
+      bool is_dir = false;
+      vfs_.is_dir(temp_dir_, &is_dir).ok();
+      if (is_dir) {
+        vfs_.remove_dir(temp_dir_).ok();
+      }
     }
+  } catch (const std::exception& e) {
+    // Suppress exceptions in destructors.
```

Comment:
CI failed because:

* When the VFS tests are cleaned-up and this destructor runs, it calls `is_dir`, which calls `ls_with_sizes`, which throws if the path does not exist[^1] (it has been deleted before, in the test).
* The destructor throws, which aborts the process.

[^1]: Should `is_dir`/`remove_dir` return false / an empty vector if the directory does not exist?

Reply:
My understanding is no, destructors should not throw any exceptions. Calling `.ok()` already has the effect of suppressing failures.